File: easy/Find-A-Writer/Find A Writer.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowWriter(line);
        }
    }
    
    static void ShowWriter(string line){
        int pos = line.IndexOf('|');
        string str =line.Substring(0,pos);
        string[] nums = line.Substring(pos+1).Trim().Split(' ');
        foreach(string num in nums){
            Console.Write(str[Convert.ToInt32(num)-1]);
        }
        Console.WriteLine();
    }
}

File: easy/Find-A-Writer/Find A Writer.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowWriter(line);
        }
    }
    
    static void ShowWriter(String line){
        int pos = line.indexOf("|");
        String str =line.substring(0,pos);
        String[] nums = line.substring(pos+1).trim().split("\\s+");
        for(String num : nums){
            System.out.print(str.charAt(Integer.valueOf(num)-1));
        }
        System.out.println();
    }
}
